File: app/src/main/java/com/bottle/track/map/model/Track.java
package com.bottle.track.map.model;

import java.io.Serializable;
import java.util.ArrayList;

import io.reactivex.annotations.NonNull;

/**
 * @Date 2017/12/5 14:54
 * @Author halfbottle
 * @Version 1.0.0
 * @Description
 */
public class Track extends Polyline implements Serializable{

    private long beginTime;     // 起始时间
    private long endTime;       // 结束时间
    private int speed;          // 速度，单位：m/s
    private int time;           // 耗时，单位：秒
    public String description;  // 描述

    public Track(@NonNull ArrayList<GeoPoint> points, long beginTime, long endTime){
        super(points);
        this.beginTime = beginTime;
        this.endTime = endTime;
        this.speed = 0; // 需要计算
        this.time = (int)(endTime - beginTime) / 1000;
    }
}

File: app/src/main/java/com/bottle/track/map/model/Track.kt
package com.bottle.track.map.model

/**
 * @ClassName Track
 * @Author half_bottle
 * @Date 2017/12/2
 * @Description Trek 中的一段轨迹，一个完整的轨迹由多段轨迹组合而成，每一段轨迹会记录距离，用时，速度等属性
 */
class Track(points: List<GeoPoint>): Polyline(points) {

    var beginTime: Long? = null   // 起始时间
    var endTime: Long? = null     // 结束时间
    var speed: Double? = null     // 速度，单位：m/s
    var time: Int? = null         // 耗时，单位：秒
    var description: String? = null

}